### PR TITLE
Remote-attestation

### PR DESCRIPTION
**Pull request by asa93**
new component in popup to display if the remote attestation of the default notary server is valid

this makes a call to the new version of tlsn-js

### CODE REVIEW

File: src/utils/constants.ts [hunk 1]
```diff
@@ -2,7 +2,8 @@ export const EXPLORER_API = 'https://explorer.tlsnotary.org';
 export const MAX_RECV = 16384;
 export const MAX_SENT = 4096;
 
-export const NOTARY_API = 'http://tlsn.eternis.ai:7047';
+export const ENCLAVE_ENDPOINT = 'https://tlsn.eternis.ai';
```
Comment (saberistic) [inline src/utils/constants.ts:5]: there are no two domains as enclave and notary, they are both part of one as https://notary.eternis.ai note that the port is default in production

File: src/remoteattestation/chain-certs.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
+export interface ChainCert {
+  Subject: string;
+  PublicKey: number[];
+  Cert: string;
+}
+
+export const chainCerts: ChainCert[] = [
```
Comment (saberistic) [inline src/remoteattestation/chain-certs.ts:7]: this variable is not used and also same is hard coded in tlsn-js.

File: src/remoteattestation/pubkey.pem [hunk 3]
```diff
@@ -0,0 +1,5 @@
+-----BEGIN PUBLIC KEY-----
```
[the remark this answers is not in the document]
Comment (saberistic) [reply src/remoteattestation/pubkey.pem:1]: looks like the file is not used at all

**Review by saberistic — COMMENTED** [review]
- I would like explanation over roles of included cert files
- looks like the nonce is hardcoded rather than randomly generated and checked

File: src/utils/constants.ts (hunk 1 above)
Comment (saberistic) [reply src/utils/constants.ts:5]: in other words considering that we are adding /enclave after reading this value that makes notary_api and enclave_endpoint as same, also we have one config in UI for this so better just merge them

File: package.json [hunk 4]
```diff
@@ -20,13 +20,20 @@
     "@extism/extism": "^1.0.2",
     "@fortawesome/fontawesome-free": "^6.4.2",
     "async-mutex": "^0.4.0",
+    "axios": "^1.7.5",
```
Comment (saberistic) [inline package.json:23]: is there an specific need for adding axios here? I imagine browser native fetch was sufficient

File: src/reducers/remote-attestation.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response = await axios.get(enclaveEndpoint);
+        setRemoteAttestation(response.data);
+        //analyze attestation validity here
+        const decoded = decodeCborAll(response.data);
+        console.log(decoded);
+
+        setIsValid(true);
```
Comment (saberistic) [inline src/reducers/remote-attestation.ts:42]: after decoding the cbor we still need to validate the certs and signatures on payload before setting this as valid, correct?

Comment (asa93) [reply src/reducers/remote-attestation.ts]: yes doing that on rust now

File: src/utils/constants.ts (hunk 1 above)
Comment (asa93) [reply src/utils/constants.ts]: fixed

File: package.json (hunk 4 above)
Comment (asa93) [reply package.json:24]: fetch is annoying for post req or when it has a body, also for error handling

**Pull request merged**